src/clnt_vc.c

```diff
@@ -312,7 +312,7 @@ clnt_vc_ncreate2(int fd,	/* open file descriptor */
 		if (ct->ct_addr.len)
 			mem_free(ct->ct_addr.buf, ct->ct_addr.len);
 
-		if (xd->refcnt == 1) {
```

Comment:
I saw that part.  What I missed was the assignment to rec->hdl.xd on the allocation line.  This looks correct.